### DISCUSSION TITLE
Threaded example in MacOS

### DISCUSSION THREAD

**Discussion by ricardobeat** [uNetworking/uWebSockets#858] [Q&A] [answered]
When running the `HelloWorldThread` example with LibUV in MacOS, it fails to initialize beyond a single thread:

```
▸ ./HelloWorldThreaded
Thread Thread Thread Thread 0x7000005fb000 failed to listen on port 30000x700000472000 listening on port 0x7000004f5000 failed to listen on port 3000

3000
0x700000578000 failed to listen on port 3000
^C
```

I noticed the uSocket library has a `OPTION_REUSE_PORT` but it doesn't seem to be used. Is `SO_REUSEADDR / REUSEPORT` necessary for the threaded example, and if so, how to enable it?

Note: had to add the `-luv` parameter to compile, otherwise it tries to use epoll despite the docs stating the contrary.

**Comment by ghost** [comment, marked as answer]
SO_REUSEPORT is only really supported on Linux. I don't think any other OS has documented similar behavior, even if it might work. I don't know the situation on macOS.
